openmc/cmfd.py

```diff
@@ -296,225 +603,2224 @@ def norm(self):
-            element = ET.SubElement(self._cmfd_file, "feeback")
-            element.text = str(self._feedback).lower()
-
-    def _create_gauss_seidel_tolerance_subelement(self):
-        if self._gauss_seidel_tolerance is not None:
-            element = ET.SubElement(self._cmfd_file, "gauss_seidel_tolerance")
-            element.text = ' '.join(map(str, self._gauss_seidel_tolerance))
-
-    def _create_ktol_subelement(self):
-        if self._ktol is not None:
-            element = ET.SubElement(self._ktol, "ktol")
-            element.text = str(self._ktol)
-
-    def _create_mesh_subelement(self):
-        if self._cmfd_mesh is not None:
-            xml_element = self._cmfd_mesh._get_xml_element()
-            self._cmfd_file.append(xml_element)
-
-    def _create_norm_subelement(self):
-        if self._norm is not None:
-            element = ET.SubElement(self._cmfd_file, "norm")
-            element.text = str(self._norm)
-
-    def _create_power_monitor_subelement(self):
-        if self._power_monitor is not None:
-            element = ET.SubElement(self._cmfd_file, "power_monitor")
-            element.text = str(self._power_monitor).lower()
-
-    def _create_run_adjoint_subelement(self):
-        if self._run_adjoint is not None:
-            element = ET.SubElement(self._cmfd_file, "run_adjoint")
-            element.text = str(self._run_adjoint).lower()
-
-    def _create_shift_subelement(self):
-        if self._shift is not None:
-            element = ET.SubElement(self._shift, "shift")
-            element.text = str(self._shift)
-
-    def _create_spectral_subelement(self):
-        if self._spectral is not None:
-            element = ET.SubElement(self._spectral, "spectral")
-            element.text = str(self._spectral)
-
-    def _create_stol_subelement(self):
-        if self._stol is not None:
-            element = ET.SubElement(self._stol, "stol")
-            element.text = str(self._stol)
-
-    def _create_tally_reset_subelement(self):
-        if self._tally_reset is not None:
-            element = ET.SubElement(self._tally_reset, "tally_reset")
-            element.text = ' '.join(map(str, self._tally_reset))
-
-    def _create_write_matrices_subelement(self):
-        if self._write_matrices is not None:
-            element = ET.SubElement(self._cmfd_file, "write_matrices")
-            element.text = str(self._write_matrices).lower()
-
-    def export_to_xml(self):
-        """Create a cmfd.xml file using the class data that can be used for an OpenMC
-        simulation.
+    @gauss_seidel_tolerance.setter
+    def gauss_seidel_tolerance(self, gauss_seidel_tolerance):
+        check_type('CMFD Gauss-Seidel tolerance', gauss_seidel_tolerance,
+                   Iterable, Real)
+        check_length('Gauss-Seidel tolerance', gauss_seidel_tolerance, 2)
+        self._gauss_seidel_tolerance = gauss_seidel_tolerance
+
+    def run(self, omp_num_threads=None, intracomm=None):
```

Comment:
Rather than having arguments for `omp_num_threads` and `intracomm`, I would just have it accept `**kwargs` that gets passed to `openmc.capi.init` -- this enables a user to pass whatever arguments (and intracommunicator) they want.

Reply:
Actually, I see that you explicitly need the intracomm because it's used directly. Nevertheless, things like the number of threads could be picked up by `**kwargs`